var/spack/repos/builtin/packages/exago/package.py

```diff
@@ -62,6 +71,46 @@ class Exago(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("cmake@3.18:", type="build")
 
+    # Profiling and optimizations
+    depends_on("hiop+deepchecking~full_optimizations", when="+hiop~full_optimizations")
+    depends_on("hiop~deepchecking+full_optimizations", when="+hiop+full_optimizations")
+
+    # Force RelWithDebInfo when not using optimizations
```

Comment:
Same comments as for HiOp in forcing a build type - not sure I have seen other packages experiment with this, so I can't really reference any best practices.